### PR TITLE
feat: initial pay with any token setup

### PR DESCRIPTION
**Pull request by JackShort**
* creates initial pay with any token setup
* dropdown does not change the input currency
* ETH is the only input currency currently for demonstration purposes
* does not affect the previous implementation

Screen Shot 2023-01-12 at 1 27 54 PM

### CODE REVIEW

File: src/nft/components/bag/BagFooter.tsx [hunk 1]
```diff
@@ -172,24 +190,55 @@ export const BagFooter = ({
...
+        {shouldUsePayWithAnyToken && (
+          <CurrencyRow>
+            <Column gap="xs">
+              <ThemedText.SubHeaderSmall textAlign="start">
+                <Trans>Pay with</Trans>
+              </ThemedText.SubHeaderSmall>
+              <CurrencyInput>
+                <CurrencyLogo currency={inputCurrency} size="24px" />
+                <ThemedText.HeadlineSmall fontWeight={500} lineHeight="24px">
+                  {inputCurrency?.symbol}
```
Comment (cbachmeier) [inline src/nft/components/bag/BagFooter.tsx:202]: Should the ETH price be aligned with the icon?
Screen Shot 2023-01-13 at 09 43 04

File: src/nft/components/bag/BagFooter.tsx [hunk 2]
```diff
@@ -172,24 +190,55 @@ export const BagFooter = ({
...
-          <Row justify="space-between">
-            <div>
-              <ThemedText.HeadlineSmall>Total</ThemedText.HeadlineSmall>
-            </div>
-            <div>
-              <ThemedText.HeadlineSmall>
+        {shouldUsePayWithAnyToken && (
+          <CurrencyRow>
+            <Column gap="xs">
+              <ThemedText.SubHeaderSmall textAlign="start">
```
Comment (cbachmeier) [inline src/nft/components/bag/BagFooter.tsx:196]: is `start` not the default?

File: src/nft/components/bag/BagFooter.tsx [hunk 3]
```diff
@@ -172,24 +190,55 @@ export const BagFooter = ({
...
+              <CurrencyInput>
+                <CurrencyLogo currency={inputCurrency} size="24px" />
+                <ThemedText.HeadlineSmall fontWeight={500} lineHeight="24px">
+                  {inputCurrency?.symbol}
+                </ThemedText.HeadlineSmall>
+                <ChevronDown size={20} color={theme.textSecondary} />
+              </CurrencyInput>
+            </Column>
+            <Column gap="xs">
+              <ThemedText.SubHeaderSmall marginBottom="4px" textAlign="end">
```
Comment (cbachmeier) [inline src/nft/components/bag/BagFooter.tsx:208]: if you move `textAlign="end"` up one level would all these components inherit it?

**Review by cbachmeier — APPROVED** [review]
LGTM minor comments

File: src/nft/components/bag/BagFooter.tsx (hunk 1 above)
Comment (JackShort) [reply src/nft/components/bag/BagFooter.tsx:202]: actually no - this is the way it is in figma

File: src/nft/components/bag/BagFooter.tsx (hunk 2 above)
Comment (JackShort) [reply src/nft/components/bag/BagFooter.tsx:196]: you right it do be

File: src/nft/components/bag/BagFooter.tsx (hunk 3 above)
Comment (JackShort) [reply src/nft/components/bag/BagFooter.tsx]: valid

**Pull request merged**